compiler/x/codegen/X86BinaryEncoding.cpp

```diff
@@ -1665,13 +1696,27 @@ TR::X86RegImmSymInstruction::addMetaDataForCodeAddress(uint8_t *cursor)
          if (getNode() && getNode()->getInlinedSiteIndex() == -1 &&
             (void *)(uintptr_t) getSourceImmediate() == cg()->comp()->getCurrentMethod()->resolvedMethodAddress())
             setReloKind(TR_RamMethod);
+         symbolKind = TR::SymbolType::typeMethod;
          // intentional fall-through
       case TR_ClassPointer:
-         cg()->addExternalRelocation(new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor,
-                                                           (uint8_t*)getNode(),
-                                                           (TR_ExternalRelocationTargetKind)getReloKind(),
-                                                           cg()),
-                                __FILE__, __LINE__, getNode());
+         if (symbolKind == TR::SymbolType::typeClass && cg()->comp()->getOption(TR_UseSymbolValidationManager))
```

Comment:
why only typeClass? are you intentionally not handling typeMethod with the SymbolManager?

Reply:
In the beginning I did try to have this work for even the other method relocation record types, but I ran into some issues that I never got around to dealing with (for reasons of time constraints); so I figured I'd leave those alone for now until the future. I can open some issues to track all of these TODOs.

Reply:
Well, ok. It seems like a strange way to structure it (switch that separates out method from class handling, but all cases default fall-through to a class-specific case that then checks if it's typeClass before using the new or the old infra). Why not put the old infra into the lowest fall-through for the method cases (reordering if needed) and the same for the newer symbol manager handling at the bottom of all the class cases?

Oh, I see, because you want to be able to selectively not use the symbol manager at all.... Ok. Somewhat yuck, but I don't see an appreciably cleaner way and it should mostly resolve itself when you figure out the method pointer issues. Also feel mildly queasy that methods are flaky but classes aren't. Are you sure it wasn't just early implementation glitches that have since been resolved? I wouldn't have thought these two fundamental ones would behave significantly differently...